cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/BQValidation.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.cloudsqlpostgresql;
+
+import com.google.cloud.bigquery.TableResult;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import org.apache.spark.sql.types.Decimal;
+import org.junit.Assert;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.sql.*;
+import java.time.*;
+import java.util.Date;
+import java.text.ParseException;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+/**
+ * BQValidation.
+ */
+public class BQValidation {
+    static List<JsonObject> BigQueryResponse = new ArrayList<>();
+    static List<Object> bigQueryRows = new ArrayList<>();
+
+    /**
+     * Extracts entire data from source and target tables.
+     *
+     * @param sourceTable table at the source side
+     * @param targetTable table at the sink side
+     * @return true if the values in source and target side are equal
+     */
+    public static boolean validateDBToBQRecordValues(String schema, String sourceTable, String targetTable)
+            throws SQLException, ClassNotFoundException, IOException, InterruptedException {
+        getBigQueryTableData(targetTable, bigQueryRows);
+        for (Object rows : bigQueryRows) {
+            JsonObject json = new Gson().fromJson(String.valueOf(rows), JsonObject.class);
```

Comment:
You have already used Lists as global variables. Create a global Gson object and re use it.

Reply:
Done :+1:

Reply:
change done